### PR TITLE
[factory] Add staking hardfork in workingset

### PR DESCRIPTION
**Pull request by Liuhaai**
# Description

This commit is to be reverted after v1.8.0.
The web3 Staking actions are prohibited to run before v1.8.0

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: state/factory/workingset.go [hunk 1]
```diff
@@ -122,6 +125,9 @@ func (ws *workingSet) runActions(
 }
 
 func withActionCtx(ctx context.Context, selp action.SealedEnvelope) (context.Context, error) {
+	if !protocol.MustGetFeatureCtx(ctx).EnableWeb3Staking && isWeb3StakingAction(selp) {
+		return nil, errUnsupport
+	}
```
Comment (dustinxie) [inline state/factory/workingset.go:130]: why check here if `runAction` will also check?

File: state/factory/workingset.go [hunk 2]
```diff
@@ -35,6 +36,8 @@ var (
 		},
 		[]string{"type"},
 	)
+
+	errUnsupport = errors.New("supported action")
```
Comment (dustinxie) [inline state/factory/workingset.go:40]: be specific `errUnsupportWeb3Staking = errors.New("unsupported web3 staking")`

File: state/factory/workingset.go [hunk 3]
```diff
@@ -151,6 +157,9 @@ func (ws *workingSet) runAction(
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, action.ErrGasLimit
 	}
+	if !protocol.MustGetFeatureCtx(ctx).EnableWeb3Staking && isWeb3StakingAction(elp) {
+		return nil, errUnsupport
+	}
```
Comment (dustinxie) [inline state/factory/workingset.go:162]: the caller needs to assert, if this happens, skip this error and continue (so the tx is silently dropped, and continue to pick other txs), similar to line 460:
```
case action.ErrChainID:
				continue
```

File: state/factory/workingset.go [hunk 4]
```diff
@@ -570,3 +579,37 @@ func (ws *workingSet) CreateBuilder(
 		SetLogsBloom(calculateLogsBloom(ctx, ws.receipts))
 	return blkBuilder, nil
 }
+
+func isWeb3StakingAction(selp action.SealedEnvelope) bool {
+	if selp.Encoding() == uint32(iotextypes.Encoding_ETHEREUM_RLP) {
+		act := selp.Action()
+		if _, ok := act.(*action.CreateStake); ok {
```
Comment (CoderZhi) [inline state/factory/workingset.go:586]:
```
    switch act.(type) {
    case *action.CreateStake, *action.DepositToStake, *action.ChangeCandidate, ...:
        return true
    default:
        return false
```

File: state/factory/workingset.go [hunk 5]
```diff
@@ -122,6 +125,9 @@ func (ws *workingSet) runActions(
 }
 
 func withActionCtx(ctx context.Context, selp action.SealedEnvelope) (context.Context, error) {
+	if !protocol.MustGetFeatureCtx(ctx).EnableWeb3Staking && isWeb3StakingAction(selp) {
```
Comment (CoderZhi) [inline state/factory/workingset.go:128]: delete this section and add error handling in https://github.com/iotexproject/iotex-core/pull/3345/files#diff-5168c619aa2e8281d3f34e0423560a0a1fda52f9fd1a49e57ba8390f0b3bfbd3R465

**Review by dustinxie — APPROVED** [review]

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**